### ISSUE TITLE
Remote caching is broken on Windows

### ISSUE THREAD

**Issue by gdeest** [tweag/rules_haskell#744]
Using `rules_haskell` with remote caching enabled is broken on Windows. Absolute paths to package databases find their way into the output of build actions, which leads to build failures when those are re-used on another machine when paths don't match.

This is likely due to lack of sandboxing and/or the various hacks we have put in place to work around the limitations of the Windows file system.

As a side remark: we currently don't test cache behavior on CI. Is this something we would want to put in place ?

**Comment by mboes** [comment]
Indeed, it's hard to be remote cache friendly when there is nothing (like sandboxing) to ensure the build is (quasi-)hermetic.

> As a side remark: we currently don't test cache behavior on CI. Is this something we would want to put in place ?

What test would you want to see?

**Comment by gdeest** [comment]
Perhaps something semi-realistic of the form of:

- Initializing an empty remote cache in a VM.
- Spawning a partial build in a first VM that populates the cache.
- Spawning another build in a different VM where tools are differently setup and paths don't match, which reuses some of the artifacts produced by the first build.

But admittedly that's a lot of setup for probably limited benefits.
